doc/mgmt/SONiC_TIMEZONE_HLD.md

```diff
@@ -0,0 +1,546 @@
+              
+
+### 1.1.2 Backend mechanisms to support timezone/clock configuration and query 
+
+#### 1.1.2.1 add/delete timezone 
+This creates or deletes the CLOCK entry in the Redis ConfigDB.
+
+```
+   "CLOCK": {
+        "system": {
+            "timezone_name": "US/Hawaii"
+        }
+    }
+```
+
+A change in the CLOCK entry triggers hostcfgd to start the CLOCK configuration script which ensures the timezone and timestamp to be consistent across host and various containers.   
+         
+For the host:   
+         
+1. change the local timezone on the host via command "timedatectl set-timezone <new_timezone>"  
+   "local time" is now modified by linking the file /etc/localtime to the /usr/share/zoneinfo/<timezone_name> directory.      
+2. retart rsyslog to apply the updated time via command "systemctl restart rsyslog.service"    
+        
+For the containers:     
+         
+1. with bind mounts, container /etc/localtime is updated when host /etc/localtime is updated. /etc/localtime points to the current timezone under /usr/share/zoneinfo/<timezone_name>.   
+   If the bind mount already exists, but to a different location in the container, then a soft link has to be made so that /etc/local is pointing to the correct timezone (e.g mgmt and telemetry)    
+2. restart rsyslog to apply the updated time via command "docker exec -it <container_name> supervisorctl restart rsyslogd"   
+       
+   For example in bgp container, once this is done, frr.log will use the updated timestamp.    
+3. For the logs not based on syslog, special handling needs to be applied.    
+       
+   For example in mgmt-framework container, rest_server.log is based on glog.   
+         
+   The transformer function needs to listen for CLOCK CONFIG DB change and updates the local timezone for glog.   
+       
+   And glog.go will have a patch to process the local timezone.       
+    
+   Below is an example of updated timestamp in the logs. The UTC offset is the difference in hours and minutes from UTC for a specific timezone. Note that the UTC offset for the same timezone will be different depending on if DST is active.      
+   ```
+   Jul 20 00:00:30.251175-07.00 2021     
+   ...
+   Nov 18 00:06:14.025528-08.00 2021  
+
+   ```
+     
+#### 1.1.2.2 set time and date 
+Details to be added. 
+    
+### 1.1.3 Functional Requirements
+
+Provide management framework support to          
+- configure timezone     
+- query timezone     
+- configure time and date    
+- query time and date     
+    
+### 1.1.4 Configuration and Management Requirements
+- CLI style configuration and show commands    
+- REST API support    
+- gNMI Support    
+           
+### 1.1.5 Configurations not supported by this feature using management framework:
+- support configuration of timezone abbreviations with offset to UTC        
+- support get timezone from dhcp option           

```

Comment:
Is DHCP option supported or not ?

Reply:
There is no plan to implement this in this release.